find/main/routes.py

```diff
@@ -195,12 +194,15 @@ def retrieve_spreadsheet(fund_code: str, submission_id: str):
         return abort(404)
 
     form = RetrieveForm()
-
+    file_metadata = file_header["metadata"]
+    file_name = file_metadata.get("download_filename")
+    if not file_name:
+        file_name = f"{fund_code}-{submission_id}"
```

Comment:
I think in the fallback case we would still need to include the file extension, eg `.xlsx`.